### PR TITLE
Migrate FSTQueryTests to C++ / remove FSTQuery

### PR DESCRIPTION
**Pull request by wilhuff**
This is the last PR in the chain that fully migrates FSTQuery to C++ Query.

### CODE REVIEW

**Comment by wilhuff** [comment]
Note the diffbase on #3484. Once that lands I'll retarget this one.

File: Firestore/core/src/firebase/firestore/model/document_set.h [hunk 1]
```diff
@@ -44,6 +44,8 @@ class DocumentComparator : public util::FunctionComparator<Document> {
 
   static DocumentComparator ByKey();
 
+  using FunctionComparator<Document>::Compare;
```
Comment (mikelehen) [inline Firestore/core/src/firebase/firestore/model/document_set.h]: Is this used?  Did you maybe mean to change line 53 below to take advantage of this?

Comment (wilhuff) [reply Firestore/core/src/firebase/firestore/model/document_set.h]: This exists to make overloading work across class scopes. See http://www.stroustrup.com/bs_faq2.html#overloadderived.

Now that I've added it line 53 can use it, so I've simplified that.

I've also added a TODO to remove the using statement once the Objective-C overload is removed.

Comment (mikelehen) [reply Firestore/core/src/firebase/firestore/model/document_set.h]: Wow.  Super awesome!

**Pull request merged**